frappe/model/base_document.py

```diff
@@ -85,6 +85,9 @@ def __init__(self, d):
 		if hasattr(self, "__setup__"):
 			self.__setup__()
 
+	def __getitem__(self, key):
+		return self.get(key)
```

Comment:
This will return `None` for non-existing keys, and propagate it till it fails. Earlier it would immediately with `AttributeError`.

Reply:
Earlier this would fail with `TypeError: '{DocType}' object is not subscriptable`. Am I missing something here? 

---

Should this be 

```suggestion
		return self.get(key) or frappe.throw(msg=key, exc=KeyError)
```

instead?

Reply:
I am not talking about subscripting. Accessing keys in general.

`doctype.key` would throw AttributeError, but now doctype["key"] will just return `None`. This is not common behaviour as `Dict` throws key error. 


The suggested change isn't fine either it will fail when values are actually `None`, failure should only be there when key doesn't exist. Maybe`getattr` is better (but that's not identical to `Document.get` 🥲 ?

Reply:
Okay, now I'm confused if we should raise KeyError or AttributeError if the attribute doesn't exist. I'm choosing `AttributeError` because although it feels like a dict, looks like a dict (sometimes), is a Document.

```suggestion
		return self.get(key) if hasattr(self, key) else frappe.throw(msg=key, exc=AttributeError)
```

Reply:
Isn't this `getattr(self, key)` with extra steps 😄

Reply:
CRAY

```suggestion
		return getattr(self, key)
```